src/commands/to_release.rs

```diff
@@ -36,23 +36,30 @@ where
         .expect("Writing to Shell doesn't fail");
 
 
-    let mut registry = RegistrySource::remote(
-        SourceId::crates_io(ws.config())
-            .expect("Your main registry (usually crates.io) can't be read. Please check your .cargo/config"),
-        &Default::default(),
-        ws.config()
-    );
+    let mut already_published = HashSet::new();
 
-    registry.update().expect("Updating from remote registry failed :( .");
+    { // to kepe the lock only around this block
+        let _lock = ws.config().acquire_package_cache_lock();
+
+        let mut registry = RegistrySource::remote(
+            SourceId::crates_io(ws.config())
+                .expect("Your main registry (usually crates.io) can't be read. Please check your .cargo/config"),
+            &Default::default(),
+            ws.config()
+        );
 
-    let mut already_published = HashSet::new();
 
-    for m in members.iter() {
-        let dep = Dependency::parse_no_deprecated(m.name(), Some(&m.version().to_string()), registry.source_id())
-            .expect("Parsing our dependency doesn't fail");
-        registry.query(&dep, &mut |_| {
-            already_published.insert(m.name());
-        }).expect("Quering the local registry doesn't fail");
+        registry.update().expect("Updating from remote registry failed :( .");
+
+
+        for m in members.iter() {
+            let dep = Dependency::parse_no_deprecated(m.name(), Some(&m.version().to_string()), registry.source_id())
+                .expect("Parsing our dependency doesn't fail");
+            registry.query(&dep, &mut |_| {
+                already_published.insert(m.name());
+            }).expect("Quering the local registry doesn't fail");
+        }
+        // drop the global package lock
```

Comment:
nit: If we're also commenting this here, we might as well do explicit `drop(_lock)` and/or and encase code here in a separate block